modules/OFConnectionManager2/module/src/cxn_instance.c

```diff
@@ -1164,13 +1167,19 @@ process_message(connection_t *cxn)
         return;
     }
 
-    ind_cxn_process_message(cxn, obj);
+    /* ignore the return value. */
+    rv = ind_cxn_process_message(cxn, obj);
+    if (rv == INDIGO_ERROR_PENDING) {
+        ind_cxn_block_async_op(cxn);
+    } 
```

Comment:
to make things more clear/less prone to weird bugs, i suggest blocking async from only one place, probably within the bundle task handler, and unblocking in the resume handler.

also, we need to document this behavior somewhere, probably in of_state_manager.h, and the appropriate cross-referencing comments should be added to bundle.c and gentable_handlers.c.  then everything can found in a small amount of time without having to search all over the place to figure out how this is intended to work.